src/Engine.php

```diff
@@ -79,13 +79,13 @@ public function remove($searchableEntities): array
             if (null === $searchableArray || 0 === \count($searchableArray)) {
                 continue;
             }
-            $indexUid = $entity->getIndexName();
+            $indexUid = $entity->getIndexUid();
 
             if (!isset($data[$indexUid])) {
                 $data[$indexUid] = [];
             }
 
-            $data[$indexUid][] = $entity->getId();
+            $data[$indexUid][] = $searchableArray['id'];
```

Comment:
Please change that back to `$entity->getId()`. With that, we (the IDE, static code analysis tools) know that inside `$data[$indexUid][]` is an `int|string` as this is the return type of  `$entity->getId()`.

Addition:
It would be great if you can add this (the comment) in line 72:
```
if ($searchableEntities instanceof SearchableEntity) {
    /** @var SearchableEntity[] $searchableEntities */
    $searchableEntities = [$searchableEntities];
}
```

With that it is let IDE discover what type `$entity` is and you get better code support.

I hope that was all clear 😄

Reply:
```suggestion
            $data[$indexUid][] = $entity->getId();
```